### PR TITLE
Refactored player code into a state machine system

### PR DESCRIPTION
**Pull request by AntGa**
Changed the player to implement a state based system, also changed the camera2d with a better formula calculation and zoomed it in by 1.5x

### CODE REVIEW

File: scripts/Camera2D.gd [hunk 1]
```diff
@@ -0,0 +1,10 @@
+extends Camera2D
+
+@export var target : Player
```
Comment (tulza) [inline scripts/Camera2D.gd]: probbaly just name this player

File: scripts/Player/player.gd [hunk 2]
```diff
@@ -4,6 +4,7 @@ extends CharacterBody2D
 const SPEED: float = 300.0
 var bullet = preload("res://scenes/bullet.tscn")
```
Comment (tulza) [inline scripts/Player/player.gd]: bullet_sprite?

File: scripts/Camera2D.gd (hunk 1 above)
Comment (AntGa) [reply scripts/Camera2D.gd]: okay done

File: scripts/Player/player.gd (hunk 2 above)
Comment (AntGa) [reply scripts/Player/player.gd]: ok

**Review by tulza — APPROVED** [review]

**Pull request merged**